tools/shared.py

```diff
@@ -2336,14 +2336,10 @@ def check_call(cmd, *args, **kw):
     raise
 
 def suffix(name):
-  parts = name.split('.')
-  if len(parts) > 1:
-    return parts[-1]
-  else:
-    return None
+  return os.path.splitext(name)[1]
```

Comment:
Hmm, yeah, strange about 1550 - it looks like it must be wrong. But we have tests on that. So it seems like that condition must be removable as it's always false..? Maybe a followup investigation, yeah.

About 338, yeah, it could be simpler I guess, but I think it's ok to leave it as is.